test/cleanup.test.js

```diff
@@ -1,187 +1,284 @@
-        committer GitHub <noreply@github.com> 1502743733 -0400 \n\
-        \n\
-        Merge pull request #15 from mapbox/categorizer';
-        return { stdout: gitMergeCommitStdout };
-      } else if(args[3] === 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' || args[3] === 'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb') {
-        t.equals(args[0], `--git-dir=${process.argv[4]}/.git`);
-        t.ok((args[1] === 'cat-file' || args[1] === 'rev-parse'), 'cat-file or rev-parse');
-        t.ok((args[2] === '-p' || args[2] === '--verify'), '-p or verify');
-        return { stdout: args[3], stderr: null };
-      } else if(args[3] === 'dddddddddddddddddddddddddddddddddddddddd') {
-        return { stdout: args[3], stderr: null };
-      }else if (args[1] === 'tag') {
-        return { stdout: null, stderr: null };
-      }
-      break;
-    case 'grep':
-      if (args[2] === gitMergeCommitStdout) {
-        return { stdout: '2', stderr: null };
-      } else if (args[1] === 'tag' && args[0] === 'dddddddddddddddddddddddddddddddddddddddd') {
-        return { stdout: 'tag', stderr: null };
-      } else if ((args[0] === 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' || args[0] === 'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb') && args[1]) {
-        t.equals(args[1], args[0]);
-        return { stdout: args[0], stderr: null };
-      } else {
-        return { stdout: null, stderr: null };
+
+
+const exec = {
+  restore: () => child_process.exec.restore(),
+
+  mock: () => {
+    return sinon.stub(child_process, 'exec', (cmd, callback) => {
+      const commandParts = cmd.split(' ');
+      const command = commandParts[2];
+
+      const sha = {
+        'cat-file': commandParts[4],
+        tag: commandParts[5],
+        'rev-parse': commandParts[4]
+      }[command];
+
+      let stdout = '';
+
+      switch (command) {
+      case 'cat-file':
+        if (sha === 'cccccccccccccccccccccccccccccccccccccccc' ||
+            sha === 'eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee') {
+          stdout = '2'; // these are merge commits
+        } else {
+          stdout = '0';
+        }
+
+        break;
+
+      case 'tag':
+        if (sha === 'dddddddddddddddddddddddddddddddddddddddd')
+          stdout = sha; // this one is a tag
+
+        break;
+
+      case 'rev-parse':
+        if (sha !== 'this-is-an-image-i-named-myself')
+          stdout = sha; // everything is a commit except this one image tag
+
+        break;
       }
 
-    }
-  };
-  require('child_process').spawnSync = spawns;
-  t.end();
-});
+      callback(null, stdout);
+    });
+  }
+};
 
-test('handleCb, error', (t) => {
-  let stub = sinon.stub(process, 'exit');
-  let log = sinon.stub(console, 'log');
-  file.handleCb(error);
-  console.log.restore();
-  t.equal(stub.getCall(0).args[0], 1, 'error should exit 1');
-  t.equal(log.getCall(0).args[0], error, 'should log error');
-  process.exit.restore();
-  t.end();
-});
+test('getImages, error', (assert) => {
```

Comment:
OCD: I prefer `assert` to `t` -- just because its better at conveying what it is.